modules/sfp_threatjammer.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:        sfp_threatjammer
+# Purpose:     Check if an IP or netblock is malicious according to ThreatJammer.com.
+#
+# Author:      diego.parrilla.santamaria@gmail.com
+#
+# Created:     2022-05-03
+# Copyright:   (c) Diego Parrilla 2022
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import time
+
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+class sfp_threatjammer(SpiderFootPlugin):
+
+    meta = {
+        'name': "Threat Jammer",
+        'summary': "Check if an IP address is malicious according to ThreatJammer.com",
+        'flags': ["apikey"],
+        'useCases': ["Passive", "Investigate"],
+        'categories': ["Reputation Systems"],
+        'dataSource': {
+            'website': "https://threatjammer.com",
+            'model': "FREE_AUTH_LIMITED",
+            'references': [
+                "https://threatjammer.com/docs/what-is-threat-jammer",
+                "https://threatjammer.com/docs/how-threat-jammer-works",
+                "https://threatjammer.com/docs/introduction-threat-jammer-user-api",
+                "https://threatjammer.com/docs/introduction-threat-jammer-report-api",
+                "https://threatjammer.com/tutorials/how-to-configure-fail2ban-in-ubuntu",
+                "https://threatjammer.com/tutorials/how-to-configure-cowrie-honeypot"
+            ],
+            'apiKeyInstructions': [
+                "https://threatjammer.com/docs/threat-jammer-api-keys",
+                "Register a new account with an email",
+                "Navigate to https://threatjammer.com/keys",
+                "The API Keys are listed under 'API Key' column"
+            ],
+            'favIcon': "https://threatjammer.com/favicon.ico",
+            'logo': "https://threatjammer.com/threatjammer-risk-score.png",
+            'description': "Threat Jammer is a service to access high-quality threat intelligence"
+            " data from a variety of sources, and integrate it into their applications with the"
+            " sole purpose of detecting and blocking malicious activity."
+        }
+    }
+
+    opts = {
+        'api_key': "",
+        'api_hostname': "dublin.api.threatjammer.com",
+        'risk_score_min': 35,
+        'checkaffiliates': True,
+    }
+
+    optdescs = {
+        'api_key': "API key",
+        'api_hostname': "User API hostname",
+        'risk_score_min': "Minimum Threat Jammer risk score",
+        'checkaffiliates': "Apply checks to affiliates?",
+    }
+
+    results = None
+    errorState = False
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+        self.errorState = False
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    def watchedEvents(self):
```

Comment:
Should this module also receive various netblock events and process those as various other similar modules do?